### ISSUE TITLE
Why node_modules is being scanned?!

### ISSUE THREAD

**Issue by Omranic** [stefanzweifel/laravel-stats#45]
For some reason when tried the command it gave a strange exception and seems to be scanning **`node_modules`** directory, but when I ignored that dir it worked fine. Exception stack:
```
<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
	<title>Sample &mdash; CKEditor</title>
	<link rel="stylesheet" href="sample.css">
</head>
<body>
	<h1 class="samples">
		CKEditor &mdash; Posted Data
	</h1>
	<table border="1" cellspacing="0" id="outputSample">
		<colgroup><col width="120"></colgroup>
		<thead>
			<tr>
				<th>Field&nbsp;Name</th>
				<th>Value</th>
			</tr>
		</thead>
	</table>
	<div id="footer">
		<hr>
		<p>
			CKEditor - The text editor for the Internet - <a class="samples" href="http://ckeditor.com/">http://ckeditor.com</a>
		</p>
		<p id="copy">
			Copyright &copy; 2003-2016, <a class="samples" href="http://cksource.com/">CKSource</a> - Frederico Knabben. All rights reserved.
		</p>
	</div>
</body>
</html>
<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
	<title>Sample &mdash; CKEditor</title>
	<link rel="stylesheet" href="sample.css">
</head>
<body>
	<h1 class="samples">
		CKEditor &mdash; Posted Data
	</h1>
	<table border="1" cellspacing="0" id="outputSample">
		<colgroup><col width="120"></colgroup>
		<thead>
			<tr>
				<th>Field&nbsp;Name</th>
				<th>Value</th>
			</tr>
		</thead>
	</table>
	<div id="footer">
		<hr>
		<p>
			CKEditor - The text editor for the Internet - <a class="samples" href="http://ckeditor.com/">http://ckeditor.com</a>
		</p>
		<p id="copy">
			Copyright &copy; 2003-2016, <a class="samples" href="http://cksource.com/">CKSource</a> - Frederico Knabben. All rights reserved.
		</p>
	</div>
</body>
</html>
PHP Warning:  Uncaught ErrorException: require(../../../../examples/server_side/scripts/ssp.class.php): failed to open stream: No such file or directory in /home/vagrant/Code/rinvex/cortex/node_modules/admin-lte/plugins/datatables/extensions/Scroller/examples/data/ssp.php:53
Stack trace:
#0 /home/vagrant/Code/rinvex/cortex/node_modules/admin-lte/plugins/datatables/extensions/Scroller/examples/data/ssp.php(53): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'require(../../....', '/home/vagrant/C...', 53, Array)
#1 /home/vagrant/Code/rinvex/cortex/node_modules/admin-lte/plugins/datatables/extensions/Scroller/examples/data/ssp.php(53): require()
#2 /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php(83): require_once('/home/vagrant/C...')
#3 /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php(65): Wnx\LaravelStats\ClassFinder->requireClassesFromFiles(Object(Symfony\Component\Finder\Finder))
#4 /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Cl in /home/vagrant/Code/rinvex/cortex/node_modules/admin-lte/plugins/datatables/extensions/Scroller/examples/data/ssp.php on line 53
PHP Stack trace:
PHP   1. {main}() /home/vagrant/Code/rinvex/cortex/artisan:0
PHP   2. Cortex\Foundation\Console\Kernel->handle() /home/vagrant/Code/rinvex/cortex/artisan:39
PHP   3. Illuminate\Console\Application->run() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:121
PHP   4. Illuminate\Console\Application->run() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Console/Application.php:88
PHP   5. Illuminate\Console\Application->doRun() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Application.php:125
PHP   6. Illuminate\Console\Application->doRunCommand() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Application.php:224
PHP   7. Wnx\LaravelStats\Commands\StatsListCommand->run() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Application.php:888
PHP   8. Wnx\LaravelStats\Commands\StatsListCommand->run() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Console/Command.php:167
PHP   9. Wnx\LaravelStats\Commands\StatsListCommand->execute() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Command/Command.php:264
PHP  10. Illuminate\Foundation\Application->call() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Console/Command.php:180
PHP  11. Illuminate\Container\BoundMethod::call() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/Container.php:549
PHP  12. Illuminate\Container\BoundMethod::callBoundMethod() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:31
PHP  13. Illuminate\Container\BoundMethod::Illuminate\Container\{closure}() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:87
PHP  14. call_user_func_array:{/home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29}() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29
PHP  15. Wnx\LaravelStats\Commands\StatsListCommand->handle() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29
PHP  16. Wnx\LaravelStats\Services\StatisticsListService->getData() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Commands/StatsListCommand.php:33
PHP  17. Wnx\LaravelStats\Services\StatisticsListService->findAndSortComponents() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Services/StatisticsListService.php:43
PHP  18. Wnx\LaravelStats\ClassFinder->getDeclaredClasses() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Services/StatisticsListService.php:65
PHP  19. Wnx\LaravelStats\ClassFinder->findAndLoadClasses() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php:38
PHP  20. Wnx\LaravelStats\ClassFinder->requireClassesFromFiles() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php:65
PHP  21. require_once() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php:83
PHP Fatal error:  Wnx\LaravelStats\ClassFinder::main(): Failed opening required '../../../../examples/server_side/scripts/ssp.class.php' (include_path='.:/usr/share/php') in /home/vagrant/Code/rinvex/cortex/node_modules/admin-lte/plugins/datatables/extensions/Scroller/examples/data/ssp.php on line 53
PHP Stack trace:
PHP   1. {main}() /home/vagrant/Code/rinvex/cortex/artisan:0
PHP   2. Cortex\Foundation\Console\Kernel->handle() /home/vagrant/Code/rinvex/cortex/artisan:39
PHP   3. Illuminate\Console\Application->run() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:121
PHP   4. Illuminate\Console\Application->run() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Console/Application.php:88
PHP   5. Illuminate\Console\Application->doRun() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Application.php:125
PHP   6. Illuminate\Console\Application->doRunCommand() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Application.php:224
PHP   7. Wnx\LaravelStats\Commands\StatsListCommand->run() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Application.php:888
PHP   8. Wnx\LaravelStats\Commands\StatsListCommand->run() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Console/Command.php:167
PHP   9. Wnx\LaravelStats\Commands\StatsListCommand->execute() /home/vagrant/Code/rinvex/cortex/vendor/symfony/console/Command/Command.php:264
PHP  10. Illuminate\Foundation\Application->call() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Console/Command.php:180
PHP  11. Illuminate\Container\BoundMethod::call() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/Container.php:549
PHP  12. Illuminate\Container\BoundMethod::callBoundMethod() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:31
PHP  13. Illuminate\Container\BoundMethod::Illuminate\Container\{closure}() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:87
PHP  14. call_user_func_array:{/home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29}() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29
PHP  15. Wnx\LaravelStats\Commands\StatsListCommand->handle() /home/vagrant/Code/rinvex/cortex/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29
PHP  16. Wnx\LaravelStats\Services\StatisticsListService->getData() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Commands/StatsListCommand.php:33
PHP  17. Wnx\LaravelStats\Services\StatisticsListService->findAndSortComponents() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Services/StatisticsListService.php:43
PHP  18. Wnx\LaravelStats\ClassFinder->getDeclaredClasses() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/Services/StatisticsListService.php:65
PHP  19. Wnx\LaravelStats\ClassFinder->findAndLoadClasses() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php:38
PHP  20. Wnx\LaravelStats\ClassFinder->requireClassesFromFiles() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php:65
PHP  21. require_once() /home/vagrant/Code/rinvex/cortex/vendor/wnx/laravel-stats/src/ClassFinder.php:83


  [Symfony\Component\Debug\Exception\FatalErrorException]
  Wnx\LaravelStats\ClassFinder::main(): Failed opening required '../../../../examples/server_side/scripts/ssp.class.php' (include_path='.:/usr/share/php')
```

**Comment by jerguslejko** [comment]
#42 solves this.

**Comment by stefanzweifel** [comment]
Closed as #42 has been merged.

By default the command looks only in the `app` directory. I could reproduce your bug of seeing the HTML code of the CKEditor by adding the `node_modules`-folder to the list of directories which should be scanned. 
The output happens because the HTML is in an example file called `posteddata.php`. (Even though we use `ob_*`-functions it still visible to the user, will look into that)

The bug of the `FatalErrorException` has been fixed with #42.
